### ISSUE TITLE
How to customize build torchscript model to be used in end devices codebase

### ISSUE THREAD

**Issue by danmalowany-allegro** [pytorch/pytorch#32690]
## 🚀 Feature
I want to compile my model to be executed in the Python/C script running on our customers computers/end devices, without the need to load the entire torch/libtorch package, but only what is needed based on the model operations.

## Motivation
Currently, the size of my ResNet model (for example) is ~100MB but it needs torch/libtorch, which requires ~1.5GB of space.
End devices (smart cameras, robots, etc.) are low in resources. R&D efforts for deployment on end devices includes a large efforts to optimize the model and reduce its size to minimum. Having my model accompanied by torch/libtorch is a difficult restriction. I am aware that the mobile community is leading the attention for similar features. However, considering modern smartphones resources, there is even a greater need for such a solution for other end devices.

## Current status
Currently i am doing this series of commands:
`
model = torchvision.models.resnet50(pretrained=True)
model.eval()
example = torch.ones(1, 3, 224, 224)
traced_model = torch.jit.trace(model, example)
ops = torch.jit.export_opnames(model)
traced_model.save('traced_model.pt')
with open('model_ops.yaml', 'w') as output:
    yaml.dump(ops, output)
`
The request is to enable building a model i can use in another python/c script without the need to load the entire torch or libtorch packages, but only what is needed based on the model operations.

## Alternatives
I am not aware of such alternatives. Will be happy to hear about them, if there are any.

**Comment by dreiss** [comment]
libtorch.so (CPU build on Linux) is only ~250 MB, so I'm not sure where the 1.5 GB number comes from.  Still, I understand that 250 MB is a lot.  We support custom builds like you're describing on Android (https://pytorch.org/mobile/android/#custom-build) and iOS (https://pytorch.org/mobile/ios/#custom-build).  I'd be willing to look at a pull request that adds this support to the non-mobile libtorch build.

**Comment by ljk53** [comment]
In addition to dreiss comment, you can build a customized libtorch using "scripts/build_mobile.sh", which works similarly to Android/iOS custom build.
It's called "build_mobile.sh" but it can run on your laptop with host gcc/clang. "_mobile" means it uses the same CMake options we used to optimize mobile build size.

**Comment by danmalowany-allegro** [comment]
@dreiss and @ljk53 - Thanks for your feedback.
I will try to start with scripts/build_mobile.sh, see how it goes and continue from there.

**Comment by danmalowany-allegro** [comment]
By the way, is there a difference between using torchscript or onnx for this purpose?

**Comment by dreiss** [comment]
Yes.  There are many structures and operators in PyTorch+TorchScript that are not supported by ONNX.
